gnd/src/main/java/com/google/android/gnd/persistence/local/room/FieldEntity.java

```diff
@@ -0,0 +1,158 @@
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.Index;
+import androidx.room.PrimaryKey;
+import com.google.android.gnd.model.form.Element;
+import com.google.android.gnd.model.form.Element.Type;
+import com.google.android.gnd.model.form.Field;
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.AutoValue.CopyAnnotations;
+import java.util.List;
+
+@AutoValue
+@Entity(
+    tableName = "field",
+    foreignKeys =
+        @ForeignKey(
+            entity = FormEntity.class,
+            parentColumns = "id",
+            childColumns = "form_id",
+            onDelete = ForeignKey.CASCADE),
+    indices = {@Index("form_id")})
+public abstract class FieldEntity {
+
+  private static final String TAG = FieldEntity.class.getName();
+
+  @CopyAnnotations
+  @NonNull
+  @PrimaryKey
+  @ColumnInfo(name = "id")
+  public abstract String getId();
+
+  @CopyAnnotations
+  @NonNull
+  @ColumnInfo(name = "element_type")
+  public abstract ElementEntityType getElementType();
+
+  @CopyAnnotations
+  @NonNull
+  @ColumnInfo(name = "field_type")
+  public abstract FieldEntityType getFieldType();
+
+  @CopyAnnotations
+  @Nullable
+  @ColumnInfo(name = "label")
+  public abstract String getLabel();
+
+  @CopyAnnotations
+  @ColumnInfo(name = "is_required")
+  public abstract boolean isRequired();
+
+  @CopyAnnotations
+  @Nullable
+  @ColumnInfo(name = "form_id")
+  public abstract String getFormId();
+
+  public static FieldEntity fromField(String formId, Type elementType, Field field) {
+    return FieldEntity.builder()
+        .setId(field.getId())
+        .setLabel(field.getLabel())
+        .setRequired(field.isRequired())
+        .setElementType(ElementEntityType.fromElementType(elementType))
+        .setFieldType(FieldEntityType.fromFieldType(field.getType()))
+        .setFormId(formId)
+        .build();
+  }
+
+  public static Element toElement(FieldEntityAndRelations fieldEntityAndRelations) {
+    return fieldEntityAndRelations.fieldEntity.getElementType().toElementType() == Type.FIELD
+        ? Element.ofField(FieldEntity.toField(fieldEntityAndRelations))
+        : Element.ofUnknown();
+  }
+
+  public static Field toField(FieldEntityAndRelations fieldEntityAndRelations) {
+    FieldEntity fieldEntity = fieldEntityAndRelations.fieldEntity;
+    Field.Builder fieldBuilder =
+        Field.newBuilder()
+            .setId(fieldEntity.getId())
+            .setLabel(fieldEntity.getLabel())
+            .setRequired(fieldEntity.isRequired())
+            .setType(fieldEntity.getFieldType().toFieldType());
+
+    List<MultipleChoiceEntity> multipleChoiceEntities =
+        fieldEntityAndRelations.multipleChoiceEntities;
+
+    if (!multipleChoiceEntities.isEmpty()) {
+      if (multipleChoiceEntities.size() > 1) {
```

Comment:
We may want to use @Embed for these since the relationship is 1:[0|1], but we can keep this for now.